crates/ruma-signatures/src/keys.rs

```diff
@@ -90,7 +87,31 @@ impl Ed25519KeyPair {
     /// generated from the private key. This is a fallback and extra validation against
     /// corruption or
     pub fn from_der(document: &[u8], version: String) -> Result<Self, Error> {
-        let oak = PrivateKeyInfo::from_der(document).map_err(Error::DerParse)?;
+        #[cfg(feature = "ring-compat")]
+        use self::compat::CompatibleDocument;
+        use pkcs8::der::Decode;
+
+        #[cfg(feature = "ring-compat")]
+        let backing: Vec<u8>;
```

Comment:
If this is required at all, it should go into the `#[cfg(feature = "ring-compat")]` block below (in which the explicit types can probably be removed too).

Reply:
It is, `PrivateKeyInfo` has a lifetime, which is bound to the borrowed bytes its given, so it needs to live on the stack somewhere until `from_pkcs8_oak` returns it.

Unless `from_pkcs8_oak` is called directly inside this block, but that'd duplicate that thrice.